server/s3_motr_reader.cc

```diff
@@ -344,6 +344,39 @@ size_t S3MotrReader::get_next_block(char **data) {
   *data = (char *)motr_rw_op_context->data->ov_buf[iteration_index];
   data_read = motr_rw_op_context->data->ov_vec.v_count[iteration_index];
   iteration_index++;
+
+  size_t length = data_read;
+  if (total_size_to_read < total_size_read + length)
+    length = total_size_to_read - total_size_read;
+  if (multipart_part_size == 0) {
+    // non-multipart-upload case. Just calculate md5 of the entire object.
+    md5crypt.Update(*data, length);
+  } else {
+    // the object was created with multipart upload.
+    // Calculate md5 checksums for each part independently.
+    size_t remaining = length;
+    // Each iteration jumps to the next part or the end of the buffer,
+    // whichever comes first.
+    while (remaining > 0) {
+      // current absolute position in object for the purpose of checksumming
+      size_t pos = total_size_read + length - remaining;
+      size_t next_part_pos;
+      if (pos / multipart_part_size >= multipart_num_of_parts - 1)
```

Comment:
I don't know the S3 coding standard.  Brackets are not required for one-liners?  

Also, the logic in the if statement is not intuitive to me.  Might be a good place for a comment.

Reply:
Commented.

Reply:
> Brackets are not required for one-liners?

I just checked this file and some other file that I have open in editor - seems like they are. Fixed.